### ISSUE TITLE
[🐞] /shopAccounts is completely broken

### ISSUE THREAD

**Issue by LorenzHetterich** [adamico/mechanical-mastery-plus#149]
### Which OS are you using?

LINUX

### Which Launcher?

PolyMC

### Describe the bug

On multiplayer servers, /shopAccounts allows you to transfer money from other peoples accounts to you:

```
/shopAccounts transfer <amount> <their-username> 1 <your-username> 1
```

Worse, you can even put others into dept, basically making them unable to play the modpack. To transfer 2billion units of money from their account to yours putting them into dept if they don't have enough:

```
/shopAccounts transfer -2000000000 <your-username> 1 <their-username> 1
```

As the amount value is limited to a signed integer, if you put people into enough depth, they cannot use the same command to get their money back (if their money is less than -pow(2, 32), they don't have enough money to transfer even a negative amount)

### To reproduce

1. Start Launcher
2. Install or Start Mechanical Mastery Plus 1.1.10 (not 100% sure about the version, PolyMC doesn't directly show)
3. Create / Load World / join server
4. use commands as described above 
5. gain money and lock other people out of ever using the shop
...

### Expected behavior

It should not be possible to transfer from other accounts than your own.
Transferring negative amounts of money should not be possible.
The amount should not be limited to a 32 bit integer (maybe that's a Minecraft limitation in which case it's fine).

### Logs

**latest.log**:
--

**crashlog**: 
--

**Comment by LorenzHetterich** [comment]
Update:

People can get out of dept by creating new accounts and also abusing the glitch, putting their other account into dept.
This also allows to create money out of thin air (create account, send it negative money, delete it).

**Comment by AmmoniumX** [comment]
I'll take accountability for this one, this was an AdminShop exploit with the /shopAccounts transfer command 😅. Fixed as of latest (-1.17.1) version.
